contracts/pods/PodFactory.sol

```diff
@@ -1,64 +1,77 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.0;
 
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
 import {ControllerV1} from "@orcaprotocol/contracts/contracts/ControllerV1.sol";
 import {MemberToken} from "@orcaprotocol/contracts/contracts/MemberToken.sol";
 import {IGnosisSafe} from "./interfaces//IGnosisSafe.sol";
 import {IPodFactory} from "./interfaces/IPodFactory.sol";
 
 import {TribeRoles} from "../core/TribeRoles.sol";
-import {OptimisticTimelock} from "../dao/timelock/OptimisticTimelock.sol";
 import {CoreRef} from "../refs/CoreRef.sol";
 import {ICore} from "../core/ICore.sol";
-import {IPodAdminGateway} from "./interfaces/IPodAdminGateway.sol";
+import {PodAdminGateway} from "./PodAdminGateway.sol";
+import {PodExecutor} from "./PodExecutor.sol";
 
 /// @dev This contract is primarily a factory contract which an admin
 /// can use to deploy more optimistic governance pods. It will create an
-/// Orca pod and deploy an optimistic timelock alongside it.
+/// Orca pod and deploy a timelock alongside it.
 ///
 /// The timelock and Orca pod are then linked up so that the Orca pod is
 /// the only proposer and executor.
 contract PodFactory is CoreRef, IPodFactory {
     /// @notice Orca controller for Pod
-    ControllerV1 public override podController;
+    ControllerV1 public immutable override podController;
 
     /// @notice Orca membership token for the pods. Handles permissioning pod members
-    MemberToken private immutable memberToken;
+    MemberToken public immutable memberToken;
 
     /// @notice Public contract that will be granted to execute all timelocks created
-    address public immutable podExecutor;
+    PodExecutor public immutable podExecutor;
 
-    /// @notice Mapping between podId and it's optimistic timelock
+    /// @notice Pod admin gateway, through which admin functionality on pods is accessed
+    PodAdminGateway public immutable podAdminGateway;
```

Comment:
We may want this to be upgradeable, would we just upgrade the factory too in that case?

Reply:
Have updated. Have removed setting the `podAdminGateway` in the `PodFactory` constructor, and instead the `PodAdminGateway` is now a dependency of the factory - it's necessary for `podAdminGateway.veto()`.

The `admin` is set on a per pod basis through the `PodConfig` passed to `factory.createOptimisticPod()`. To upgrade the admin:

1. New PodAdmin contract is deployed
2. TribalCouncil deploys a new pod by calling `podFactory.createOptimisticPod()`, passing this new podAdmin
3. TribalCouncil authorises new pod
4. TribalCouncil revokes and deprecates the old pod

Only the podAdmin is able to migrate a pod's admin, and even if we were to add that functionality we would need a a contract with the `TIMELOCK_ADMIN_ROLE`(maybe the `podFactory`)? For this first version of governance propose running with above flow and iterating if need be. Admin functionality is expected to be fairly stable.